tests/test_bulk_module.py

```
@@ -0,0 +1,205 @@
+    dialog._bulk_create_in_progress = True
+    dialog.reject()
+    assert calls == []  # base QDialog.reject() must not run while busy
+
+    dialog._bulk_create_in_progress = False
+    dialog.reject()
+    assert calls == [1]
+
+
+def test_bulk_create_still_works_end_to_end(qapp, tmp_path, monkeypatch):
+    # Sanity check that the guard doesn't break the ordinary happy path.
+    dialog, job_module = _make_dialog(qapp, tmp_path)
```

Ruff (0.16.1)

[warning] Unpacked variable `job_module` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)